File: src/main/java/me/zero/jarpwner/transform/Transformer.java
package me.zero.jarpwner.transform;

/**
 * Implementation of {@link ITransformer} that has a constructor which takes in a single
 * {@link ITransformerContext} argument, which may then be referenced through {@code protected}
 * access by implementors in order to access information which the context provides.
 *
 * @see ITransformerContext
 *
 * @author Brady
 * @since 4/4/2019
 */
public abstract class Transformer implements ITransformer {

    /**
     * The transformer context provided by the transform sequence
     */
    protected final ITransformerContext context;

    public Transformer(ITransformerContext context) {
        this.context = context;
    }
}

File: src/main/kotlin/me/zero/jarpwner/transform/Transformer.kt
package me.zero.jarpwner.transform

import org.objectweb.asm.tree.ClassNode

/**
 * @author Brady
 * @since 3/30/2019
 */
interface Transformer {

    fun apply(cn: ClassNode)

    fun accepts(path: String): Boolean

    fun getInfo() : Collection<String>
}
